src/Servers/Kestrel/test/InMemory.FunctionalTests/Http2/Http2ConnectionTests.cs

```diff
@@ -208,6 +210,84 @@ public async Task RequestHeaderStringReuse_MultipleStreams_KnownHeaderReused()
             await StopConnectionAsync(expectedLastStreamId: 3, ignoreNonGoAwayFrames: false);
         }
 
+        [Fact]
+        public async Task ResponseTrailers_MultipleStreams_Reset()
+        {
+            IEnumerable<KeyValuePair<string, string>> requestHeaders = new[]
+            {
+                new KeyValuePair<string, string>(HeaderNames.Method, "GET"),
+                new KeyValuePair<string, string>(HeaderNames.Path, "/hello"),
+                new KeyValuePair<string, string>(HeaderNames.Scheme, "http"),
+                new KeyValuePair<string, string>(HeaderNames.Authority, "localhost:80"),
+                new KeyValuePair<string, string>(HeaderNames.ContentType, "application/json")
+            };
+
+            var requestCount = 0;
+            await InitializeConnectionAsync(context =>
+            {
+                requestCount++;
+
+                var trailers = context.Features.Get<IHttpResponseTrailersFeature>().Trailers;
```

Comment:
But what if you do this instead?

```suggestion
                var trailers = new HttpResponseTrailers();
                context.Features.Get<IHttpResponseTrailersFeature>().Trailers = trailers;
```

Reply:
If you replace the trailers collection then it doesn't output the trailers at all

Reply:
The only reason to replace the trailers would be with a wrapper. But to the original question, would the trailers field get reset between requests?

Reply:
Updated test with a wrapper. The collection is reset, but the wrapper set in the previous request hangs around.

I'm guessing we don't want that. Create an issue for it?

Reply:
https://github.com/dotnet/aspnetcore/issues/34411